tests/observability/metrics/conftest.py

```
@@ -619,3 +625,63 @@ def aaq_resource_hard_limit_and_used(application_aware_resource_quota):
+    agent_version = output[0].strip() if output else ""
+    LOGGER.info(f"qemu-guest-agent version: {agent_version}")
+
+    # Parse version to ensure it's at least 9.6
+    version_match = re.search(r"qemu-guest-agent-(\d+)\.(\d+)", agent_version)
+    if version_match:
+        version_num = float(f"{version_match.group(1)}.{version_match.group(2)}")
+
+        LOGGER.info(f"Parsed qemu-guest-agent version: {version_num}")
+        if version_num >= 9.6:
+            return
+        raise ValueError(f"qemu-guest-agent version {version_num} is less than required 9.6")
```

Ruff (0.14.2)

Avoid specifying long messages outside the exception class

(TRY003)